include/boost/url/url_view.hpp

```diff
@@ -196,9 +196,26 @@ class BOOST_SYMBOL_VISIBLE url_view
             @ref parse_uri_reference.
     */
     BOOST_URL_DECL
-    explicit
     url_view(string_view s);
 
+    /// @copydoc url_view(string_view)
+    template<
+        class String
+#ifndef BOOST_URL_DOCS
+        , class = typename std::enable_if<
+            std::is_convertible<
+                String,
+                string_view
+                    >::value>::type
+#endif
+    >
+    url_view(
+        String const& s)
```

Comment:
now you have two overloads which are the same no? string_view is convertible to string_view. Does this compile?
```
string_view s = "";
url_view u( s );
```

Isn't this ambiguous?

Reply:
It works fine. It's exactly the same pattern as pct_string_view:

- the compiler picks up the `string_view` overload because it's not templated
- the compiler picks up the `String` overload when it's convertible to `string_view` but not `string_view`

If all things were directly convertible to `string_view`, we would just need the first overload. But we have two implicit conversion steps sometimes.

I included your test to ensure that's OK. The test is somewhat redundant because most other tests already construct these `url_view`s from `string_view`s, but that's not explicit..